### PR TITLE
trainee edit status of subject within course

### PR DESCRIPTION
**Pull request by duypn-2245**

### CODE REVIEW

**Comment by duypn-2245** [comment]
https://edu-redmine.sun-asterisk.vn/issues/25794

**Comment by duypn-2245** [comment]
image
image

**Comment by duypn-2245** [comment]
ready

File: app/controllers/user_subjects_controller.rb [hunk 1]
```diff
@@ -53,4 +62,8 @@ def check_status_closed_subject
   def load_task_ids
     @task_ids = Subject.find_by(id: @course_detail.subject_id).tasks.ids
   end
+
+  def load_user_subject
+    @user_subject = current_user.user_subjects.find_by subject_id: params[:subject_id]
```
Comment (vanvtt-0952) [inline app/controllers/user_subjects_controller.rb:72]: Xử lý nil bé

File: app/controllers/user_subjects_controller.rb [hunk 2]
```diff
@@ -17,19 +19,26 @@ def create
     redirect_to @course
   end
 
+  def update
+    @user_subject.update status: params[:status]
```
Comment (vanvtt-0952) [inline app/controllers/user_subjects_controller.rb]: thêm xử lý message fail khi update

File: app/views/process_tasks/update.js.erb [hunk 3]
```diff
@@ -1,2 +1,3 @@
 $('show_task').html('<%= j render @tasks %>')
-alert(I18n.t("process_tasks.update.success"))
+$('.btn_update').removeAttr('disabled');
+alert(I18n.t('process_tasks.update.success'))
```
Comment (vanvtt-0952) [inline app/views/process_tasks/update.js.erb:3]: .

File: app/views/user_subjects/update.js.erb [hunk 4]
```diff
@@ -0,0 +1,3 @@
+$('#show_subject').html('<%= j render "courses/show_subject" %>')
```
Comment (vanvtt-0952) [inline app/views/user_subjects/update.js.erb:1]: .

**Comment by duypn-2245** [comment]
ready

**Pull request merged**